server/main.cpp

```diff
@@ -58,22 +59,22 @@ int main()
 
              for (int i = 0; i < NUM_PLAYERS; i++)
              {
-                 if (events[i] != -1)
+                 if (events[serv->ids[i]] != -1)
                  {
                      continue; // continue if we already receive 1 event from this client
                  }
-                 int check_event = serv->recv_event(i);
-                 events[i] = check_event;
+                 int check_event = serv->recv_event(serv->ids[i]);
+                 events[serv->ids[i]] = check_event;
              }
              end = std::chrono::duration_cast<std::chrono::milliseconds>(
                  std::chrono::system_clock::now().time_since_epoch()
              );
              //end = glfwGetTime();
          }
-         serv->updateByEvent((EventType)events[0], (EventType)events[1], (EventType)events[2], (EventType)events[3]);
+         serv->updateByEvent((EventType)events[serv->ids[0]], (EventType)events[serv->ids[1]], (EventType)events[serv->ids[2]], (EventType)events[serv->ids[3]]);
```

Comment:
For the updateByEvent function, the first event corresponds to the event for Alice, so need to pass in the events related to serv->ids[i] == 0